### ISSUE TITLE
Ad block install failed

### ISSUE THREAD

**Issue by kx-chen** [StreisandEffect/streisand#1646]
### Expected behavior:
Ad blocking installs properly.

### Actual Behavior:
Fails with following error: 
```
fatal: [157.245.189.107]: FAILED! => {"msg": "The conditional check 'streisand_ad_blocking_enabled' failed. The error was: error while evaluating conditional (streisand_ad_blocking_enabled): 'y' is undefined\n\nThe error appears to be in '/Users/kai/vpn/streisand/playbooks/roles/ad-blocking/tasks/main.yml': line 1, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"Install blocklist tools\"\n  ^ here\n"}
```
### Steps to Reproduce:
I used the fresh digital ocean droplet created by streisand and installed with most options default with the exception of adblocking.

### Ansible Information

* Ansible version: 2.8.4
* Ansible system: Darwin
* Host OS: MacOSX
* Host OS version:  10.14.6
* Python interpreter: python
* Python version: 2.7.10

### Streisand Information

* Streisand Git revision: e0680b71e13c5549145899cd534a7032e885311c
* Streisand Git clone has untracked changes: no
* Genesis role: genesis-digitalocean
* Custom SSH key: False

### Enabled Roles

* Shadowsocks enabled:  True
* Wireguard enabled: True
* OpenVPN enabled: True
* stunnel enabled: True
* Tor enabled: False
* Openconnect enabled: True
* TinyProxy enabled: True
* SSH forward user enabled: True
* Configured number of VPN clients: 10

**Comment by nopdotcom** [comment]
My first guess is that something is being picky about `y` vs `yes`. Try using full `yes` and `no` and see if that works for you.

If it doesn't, maybe we have some other kind of skew. Could you run this for me?

```sh
$ ansible-config dump --only-changed
ANSIBLE_PIPELINING(/Users/nop/gh/nop-streisand/ansible.cfg) = True
DEFAULT_HOST_LIST(/Users/nop/gh/nop-streisand/ansible.cfg) = [u'/Users/nop/gh/nop-streisand/inventories/inventory']
DEFAULT_LOCAL_TMP(/Users/nop/gh/nop-streisand/ansible.cfg) = /Users/nop/.ansible/tmp/ansible-local-95409ek9gXJ
DEFAULT_MODULE_PATH(/Users/nop/gh/nop-streisand/ansible.cfg) = [u'/Users/nop/gh/nop-streisand/library']
DEFAULT_TIMEOUT(/Users/nop/gh/nop-streisand/ansible.cfg) = 100
HOST_KEY_CHECKING(/Users/nop/gh/nop-streisand/ansible.cfg) = False
$ 
```

**Comment by kx-chen** [comment]
Thanks, I'll try with using `yes` later today and let you know. :)

**Comment by holderad** [comment]
Streisand Install failed on google cloud instance after enabling customization and using all defaults except "yes" to ad block. At failure, long red error message that ultimately said NGINX failed to start. Is there a problem with either retrieving the blocked domains for DNSMASQ or perhaps the list is bad?

**Comment by holderad** [comment]
Streisand install failed on google cloud instance after enabling customization and attempting to enable ad block and OpenVPN. At failure, long pause again at NGINX and then "failed to install."

After some thought, I would recommend eliminating the ability to enable ad block and eliminate the ability to add Cloudflared. I love the project and it is awesome. 

However, I think it is suffering from feature creep. 

What's the objective here? I think it is to produce a hardened server that provides any noob the ability to punch through firewalls and allow knowledge to flow. 

Please keep it pure. The more pure, the easier it will be to maintain, the easier it will be to fix problems, and the easier it will be for noobs like myself to understand and not fear using.

Anyway...
I can always add PiHole or AdGuard Home in the background to listen on tun0. AdGuard Home in the background will do DoH upstream with ad blocking so there is no reason to dilute the StreisandEffect with trying to include it.

Sorry for the rant. This project is by far one of the best things I have ever seen. If time is the currency of life then knowledge is its blood.

 By the way, I would be interested in helping write some documentation so that noobs like myself can understand and install this awesome project. It took me a week of experimenting, googling, and blowing away instances until I got the hang of it. You will laugh, but I spent a whole day just trying to figure out how to get the dang .html file produced at the end of the setup on to my Mac so I could use it. (the gear at the top right). Those guys at google must think I nuts blowing away so many instances...haha

**Comment by jademonkey33** [comment]
I was able to get this to work on a DO droplet.  My configuration options were slightly different;

- Shadowsocks enabled: False
- Wireguard enabled: True
- OpenVPN enabled: True
- stunnel enabled: True
- Tor enabled: False
- Openconnect enabled: False
- TinyProxy enabled: True
- SSH forward user enabled: False
- Configured number of VPN clients: 5

I also explicitly used 'yes' and 'no' during any prompt entry.  My ansible version is the same as the original poster, but my python version is 2.7.12

I will attempt to spin up another instance to mirror the configuration choices of the initial poster to isolate variables later today.

**Comment by changchichung** [comment]
> Streisand install failed on google cloud instance after enabling customization and attempting to enable ad block and OpenVPN. At failure, long pause again at NGINX and then "failed to install."

yes , I have the same problem , install hangs longtime at nginx  then failed.
I had enable ad block , but no OpenVPN.

so I run again , disabled ad block , and the script works.

hope that problem can be fixed.

**Comment by changchichung** [comment]
according to the ansible error messages , nginx install failed due to temporally dns resolve error , and nameserver was assigned to 127.0.0.1 in /etc/resolv.conf 
so I add this task in stresiand/playbooks/roles/nginx/task/main.yml 
it's ugly , but it's working.

```
- name: replace nameserver in /etc/resolv.conf , or nginx will hang like forever and failed
  lineinfile:
    path: /etc/resolv.conf
    regexp: 'nameserver'
    line: 'nameserver 8.8.8.8'                                                                 
    state: present
```
now , I can run the script with ad-block enabled .

**Comment by TheNomad11** [comment]
@holderad Ad-blocking is optional and the default is No, that is a good compromise, I think. It worked well with a VPS and I am amazed by and thankful for this feature. Installing Pi-Hole can be a hassle (especially combined with Streisand, I only made it work by de-installing nginx because of conflicts).

**Comment by EntropiaFox** [comment]
@changchichung 's tweak did the trick while installing Streisand with ad-blocking enabled-- I had been racking my brains for an entire day trying to figure out what the exact issue was, to the point of even trying a different VPS.
